posthog/management/commands/test/test_backfill_distinct_id_overrides.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+import uuid
+from unittest import mock
+
+from clickhouse_driver.client import Client
+from clickhouse_driver.errors import ErrorCodes, ServerException
+from posthog.client import sync_execute
+from posthog.management.commands.backfill_distinct_id_overrides import BackfillQuery, execute_backfill
+from posthog.models.person.util import create_person_distinct_id
+from posthog.test.base import BaseTest, ClickhouseTestMixin, _create_event, flush_persons_and_events
+
+
+class ExecuteBackfillTestCase(ClickhouseTestMixin, BaseTest):
+    def __run_test_backfill(self, dry_run: bool) -> None:
+        distinct_ids_to_person_id = {
+            "no-override": uuid.uuid4(),
+            "needs-override": uuid.uuid4(),
+        }
+
+        for _ in range(3):
+            _create_event(
+                team=self.team,
+                event="invalid",
+                distinct_id="no-override",
+                person_id=distinct_ids_to_person_id["no-override"],  # keep correct person id
+            )
+
+        for _ in range(3):
+            _create_event(
+                team=self.team,
+                event="invalid",
+                distinct_id="needs-override",
+                person_id=str(uuid.uuid4()),  # mismatched value causes a backfill row
+            )
+
+        flush_persons_and_events()
+
+        for distinct_id, person_id in distinct_ids_to_person_id.items():
+            create_person_distinct_id(
+                team_id=self.team.pk,
+                distinct_id=distinct_id,
+                person_id=str(person_id),
+                version=1,
+            )
```

Comment:
This will need to detach the materialized view once #20349 is ready, otherwise test this will start failing or flaking because the version 1 written to overrides will override the version -1.